backend/inspirehep/hal/utils.py

```diff
@@ -0,0 +1,176 @@
+                "first_name": first_name,
+                "last_name": last_name,
+            }
+        )
+
+    return result
+
+
+def get_conference_record(record, default=None):
+    """Return the first Conference record associated with a record.
+    Queries the database to fetch the first Conference record referenced
+    in the ``publication_info`` of the record.
+    Args:
+        record(InspireRecord): a record.
+        default: value to be returned if no conference record present/found
+    Returns:
+        InspireRecord: the first Conference record associated with the record.
+    Examples:
+        >>> record = {
+        ...     'publication_info': [
+        ...         {
+        ...             'conference_record': {
+        ...                 '$ref': '/api/conferences/972464',
+        ...             },
+        ...         },
+        ...     ],
+        ... }
+        >>> conference_record = get_conference_record(record)
+        >>> conference_record['control_number']
+        972464
+    """
+    pub_info = get_value(record, "publication_info.conference_record[0]")
+    if not pub_info:
+        return default
+
+    conferences = get_db_records([("con", get_recid_from_ref(pub_info))])
+    return conferences[0]
+
+
+def get_divulgation(record):
+    """Return 1 if a record is intended for the general public, 0 otherwise.
+    Args:
+        record(InspireRecord): a record.
+    Returns:
+        int: 1 if the record is intended for the general public, 0 otherwise.
+    Examples:
+        >>> get_divulgation({'publication_type': ['introductory']})
+        1
+    """
+    return 1 if "introductory" in get_value(record, "publication_type", []) else 0
+
+
+def get_domains(record):
+    """Return the HAL domains of a record.
+    Uses the mapping in the configuration to convert all INSPIRE categories
+    to the corresponding HAL domains.
+    Args:
+        record(InspireRecord): a record.
+    Returns:
+        list(str): the HAL domains of the record.
+    Examples:
+        >>> record = {'inspire_categories': [{'term': 'Experiment-HEP'}]}
+        >>> get_domains(record)
+        ['phys.hexp']
+    """
+    terms = get_value(record, "inspire_categories.term", default=[])
+    mapping = current_app.config["HAL_DOMAIN_MAPPING"]
+
+    return [mapping[term] for term in terms]
+
+
+def _get_hal_id_map(record):
+    affiliations = get_value(record, "authors.affiliations.record", default=[])
+    affiliation_list = chain.from_iterable(affiliations)
+    affiliation_recids = [get_recid_from_ref(el) for el in affiliation_list]
+
+    pids = [("ins", pid) for pid in affiliation_recids]
+    institutions = get_db_records(pids)
+
+    return {el["control_number"]: _get_hal_id(el) for el in institutions}
+
+
+def _get_hal_id(record):
+    for el in record.get("external_system_identifiers", []):
+        if el.get("schema") == "HAL":
+            return el["value"]
+
+
+def get_db_records(pids):
```

Comment:
this function was used when HAL push was outside of INSPIRE as there was nothing better. You should remove this and use `InspireRecord.get_records_by_pids` instead.